src/radix_connect/app_link_parser.rs

```diff
@@ -0,0 +1,154 @@
+use crate::prelude::*;
+use url::form_urlencoded;
+use url::Url;
+
+use super::{DappRequest, LinkRequest, MobileConnectRequest};
+
+#[uniffi::export]
+pub fn parse_request_url(url: String) -> Result<MobileConnectRequest> {
+    let connect_url = Url::parse(CONNECT_URL).unwrap();
+    match Url::parse(url.as_str()) {
+        Ok(parsed_url) => {
+            if parsed_url.host_str() != connect_url.host_str()
+                || parsed_url.scheme() != connect_url.scheme()
+            {
+                return Err(CommonError::RadixConnectMobileInvalidRequestUrl);
+            }
+
+            let query_parameters = parsed_url
+                .query_pairs()
+                .into_owned()
+                .collect::<HashMap<String, String>>();
+            if let Some(session_id) =
```

Comment:
same here, something like this would remove nesting
```
 let session_id = query_parameters.get(CONNECT_URL_PARAM_SESSION_ID)
        .ok_or(CommonError::RadixConnectMobileInvalidRequestUrl)?
```

Reply:
:D we wrote the same thing :D

Reply:
at the same time :D